File: LinkedList/LinkedList Java/linkedList.java

public class linkedList {
    public class Node {
        int data;
        Node prev;
        Node next;

        public Node(int data) {
            this.data = data;
            prev = null;
            next = null;
        }
    }

    Node head;
    Node tail;
    int size;

    public linkedList() {
        head = null;
        tail = null;
        size = 0;
    }

    public String toString() {
        String listString = "[";

        Node currentNode = head;
        while (currentNode != null) {
            listString += currentNode.data;
            currentNode = currentNode.next;

            if (currentNode != null) {
                listString += ", ";
            }
        }

        listString += "]";

        return listString;
    }

    public void push(int data) {
        if (size == 0) {
            Node newNode = new Node(data);
            head = newNode;
            tail = newNode;
        } else {
            Node newNode = new Node(data);
            tail.next = newNode;
            newNode.prev = tail;
            tail = newNode;
        }
        size++;
    }

    public int peek() {
        return tail.data;
    }

    public static void main(String[] args) {
        linkedList mylis = new linkedList();
        for (int i = 0; i < 100; i++) {
            mylis.push(i);
        }
        System.out.println(mylis.peek());
    }
}

File: LinkedList/LinkedList C++/LinkedList.cpp
#include "LinkedList.hpp"

LinkedList::LinkedList() {
    size = 0;
    head = 0;
    tail = 0;
}

int LinkedList::peek() {
    return tail->data;
}

int LinkedList::pop() {
    int popData = tail->data;
    tail = tail->prev;
    delete tail->next;
    tail->next = nullptr;
    size--;
    return popData;
}

void LinkedList::push(int data) {
    if (size == 0) {
        Node newNode = Node(data);
        head = &newNode;
        tail = &newNode;
    } else if (size == 1) {
        Node newNode = Node(data);
        tail->next = &newNode;
        newNode.prev = tail;
        tail = &newNode;
        head->next = tail->prev;
    }
    {
        Node newNode = Node(data);
        tail->next = &newNode;
        newNode.prev = tail;
        tail = &newNode;
    }
    size++;
}

std::ostream &operator<<(std::ostream &os, const LinkedList &list) {
    os << "[";
    Node *currentNode = list.head;
    while (currentNode != nullptr) {
        os << currentNode->data;
        currentNode = currentNode->next;

        if (currentNode != nullptr) {
            os << ", ";
        }
    }
    os << "]";

    return os;
}